### ISSUE TITLE
sensors_temp should not use hysteresis (when zero) as warning

### ISSUE THREAD

**Issue by JosefRypacek** [munin-monitoring/munin#1346]
**Describe the bug**
Hi all,

my server has following output of sensors command. 
Critical value (80) is used as critical limit in munin.
Hyst value (0) is also used in munin as warning limit, thus warning is still active.

In my opinion this value should not be used as warning. At least when it's 0.

`# sensors
amdgpu-pci-0900
Adapter: PCI adapter
vddgfx:           N/A
vddnb:            N/A
edge:         +33.0°C  (crit = +80.0°C, hyst =  +0.0°C)`

**Comment by shtrom** [comment]
I think the problem is due to what output the plugin tries to parse. A bigger example of this problem is when `sensors` actually uses multiple lines to show all the limits and characteristics of the sensor, e.g.,

```
w83795adg-i2c-1-2f
Adapter: SMBus PIIX4 adapter port 2 at 0b00
...
MB Temp:      +22.5°C  (high = +39.0°C, hyst = +39.0°C)
                       (crit = +44.0°C, hyst = +44.0°C)  sensor = thermistor
```

This multi-line input completely confuses the plugin, which doesn't end up setting the right limits.

Field | Internal name | Type | Warning | Critical | Info
-- | -- | -- | -- | -- | --
...
MB Temp | temp5 | gauge | 39.0 | 39.0

I think the fix should be around using raw output for `sensors`:

```
$ sensors -u | grep '\(MB Temp\|temp5\)' 
MB Temp:
  temp5_input: 22.500
  temp5_max: 39.000
  temp5_max_hyst: 39.000
  temp5_crit: 44.000
  temp5_crit_hyst: 44.000
  temp5_alarm: 0.000
  temp5_type: 4.000
```

**Comment by sumpfralle** [comment]
I have to admit, that I do not really grasp the problem and the suitable solution. Could one of you give me a hint or suggest a change? Otherwise I will try to dive into it ...

**Comment by JosefRypacek** [comment]
I think there are two problems. My problem could be solved by: "Do not set warning limit if hyst value is 0.0". 

Problem reported by shtrom is differend and I don't have idea what the output is. Is it one sensor or multiple sensors? Why sensor's output has multiple lines? This can't be probably solved without explaining HW and sensors.

**Comment by shtrom** [comment]
I think the issue initially reported here is due to the plugin mis-parsing the default output format of `sensors`. It seems to default to a human-friendly output that can be broken on multiple lines, and only shows relevant fields.

By contrast, with the `-u` flag, `sensors` output machine-readable `key: value`-formatted data. I think the plugin could parse this output much more easily, and avoid misinterpretation, like this bug.

Sensors seem to have a `max` and a `crit`, which I think maps to Munin's `warning` and `critical`, it could just be a matter of `sed`ing a bit, in a much more regular way than the current parsint. I don't think the `hyst`eresis is something that munin supports at the moment, so I'd suggest to ignore it.

**Comment by JosefRypacek** [comment]
I agree, `hyst` should not be used (at least when value is 0). I know what hysteresis is and don't know why it's used as warning limit. Maybe it has "similar" meaning for some temperature sensors on some systems.
Below are sample outputs from my server.

```
# sensors
edge:         +35.0°C  (crit = +80.0°C, hyst =  +0.0°C)


# sensors -u
edge:
  temp1_input: 35.000
  temp1_crit: 80.000
  temp1_crit_hyst: 0.000


# munin-run sensors_temp config
temp1.label edge
temp1.warning 0.0
temp1.critical 80.0
```

**Comment by shtrom** [comment]
> I know what hysteresis is and don't know why it's used as warning limit. Maybe it has "similar" meaning for some temperature sensors on some systems.

I think it just gets incorrectly parsed as the warning value, just because it is the second number found on the line.

**Comment by JosefRypacek** [comment]
Just to be sure, I'm talking about plugin which source is probably at
https://github.com/munin-monitoring/munin/blob/stable-2.0/plugins/node.d.linux/sensors_.in

I don't understand perl, so not sure where the problem is.
My munin-plugins-core version is 2.0.56-1ubuntu1

**Comment by shtrom** [comment]
On 14 September 2020 11:48:55 pm AEST, "Josef Rypáček" <notifications@github.com> wrote:
>Just to be sure, I'm talking about plugin which source is probably at
>https://github.com/munin-monitoring/munin/blob/stable-2.0/plugins/node.d.linux/sensors_.in
>
>I don't understand perl, so not sure where the problem is.
>My munin-plugins-core version is 2.0.56-1ubuntu1

Yep, same (:
-- 
Olivier Mehani <shtrom@ssji.net>
Sent from my mobile. Please excuse my brevity.
